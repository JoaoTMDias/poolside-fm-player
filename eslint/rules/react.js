/**
 * React specific rules
 *
 * @author Jo√£o Dias (joao.dias@wit-software.com)
 * @since 1.0.0
 */
module.exports = {
	extends: ['react-app', 'airbnb'],
	plugins: ['react', 'react-hooks'],
	rules: {
		'react/jsx-wrap-multilines': 0,
		'react/jsx-closing-bracket-location': 0,
		'react/jsx-curly-spacing': 0,
		'react/jsx-equals-spacing': 0,
		'react/jsx-first-prop-new-line': 0,
		'react/jsx-indent': 0,
		'react/jsx-indent-props': 0,
		'react/jsx-max-props-per-line': 0,
		'react/jsx-tag-spacing': 0,
		'react/jsx-closing-tag-location': 0,
		'react/jsx-filename-extension': [
			'error',
			{
				extensions: ['.js', '.jsx', '.ts', '.tsx'],
			},
		],
		'react-hooks/rules-of-hooks': 'error',
		'react-hooks/exhaustive-deps': 'warn',
		'react/prop-types': 'off',
		'react/require-default-props': 'off',

		'jsx-a11y/href-no-hash': 'off',
		'jsx-a11y/anchor-is-valid': [
			'warn',
			{
				aspects: ['invalidHref'],
			},
		],
	},
};
