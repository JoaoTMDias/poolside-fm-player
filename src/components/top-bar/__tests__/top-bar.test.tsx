import React from "react";
import { render, cleanup } from "@testing-library/react"

import TopBar from "components/top-bar/index";

afterEach(cleanup);

describe("<TopBar />", () => {
	it("should render a top-bar", () => {
		const component = render(<TopBar />);

		expect(component).toMatchSnapshot();
	});
});
