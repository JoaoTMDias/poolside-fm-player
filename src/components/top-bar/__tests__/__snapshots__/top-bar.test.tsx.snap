// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should render a top-bar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-AxjAm jffeXQ row"
        id="top-bar"
      >
        <button
          class="top-bar__button button"
          data-testid="component-top-bar-button"
          type="button"
        >
          <span
            class="sr-only"
          >
            Close this window
          </span>
        </button>
        <div
          class="top-bar__handle"
          role="presentation"
        >
          <span
            class="top-bar__handle__item"
          />
          <span
            class="top-bar__handle__item"
          />
          <span
            class="top-bar__handle__item"
          />
          <span
            class="top-bar__handle__item"
          />
          <span
            class="top-bar__handle__item"
          />
          <span
            class="top-bar__handle__item"
          />
        </div>
        <h2
          class="top-bar__title"
        >
          Poolside FM
        </h2>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-AxjAm jffeXQ row"
      id="top-bar"
    >
      <button
        class="top-bar__button button"
        data-testid="component-top-bar-button"
        type="button"
      >
        <span
          class="sr-only"
        >
          Close this window
        </span>
      </button>
      <div
        class="top-bar__handle"
        role="presentation"
      >
        <span
          class="top-bar__handle__item"
        />
        <span
          class="top-bar__handle__item"
        />
        <span
          class="top-bar__handle__item"
        />
        <span
          class="top-bar__handle__item"
        />
        <span
          class="top-bar__handle__item"
        />
        <span
          class="top-bar__handle__item"
        />
      </div>
      <h2
        class="top-bar__title"
      >
        Poolside FM
      </h2>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
